packages/cli-core/API.md

```diff
@@ -15,6 +15,9 @@ export class AmplifyPrompter {
         message: string;
         defaultValue?: string;
     }) => Promise<string>;
+    static requiredInput: (options: {
+        message: string;
+    }) => Promise<string>;

```

Comment:
how about
```
    static input: (options: {
        message: string;
        defaultValue?: string;
        required?: boolean
    }) => Promise<string>;
```

above?

Reply:
(or some type union for `options` to make default and required mutually exclusive)

Reply:
combining it was making it slightly cryptic without making a break in the API.

e.g.
```ts
{
    message: string;
    required: false;
    defaultValue?: string;
  } | {
    message: string;
    required: true;
  }
```
is a breaking change and
```ts
{
    message: string;
    defaultValue?: string;
  } | {
    message: string;
    required: true;
  }
```
allows both `defaultValue` and `required` shows up in the intellisense which is confusing API
```ts
await AmplifyPrompter.input({
    message: `Hit [enter] when complete`,
    required: true,
    defaultValue: ''
});
```
I figured having an explicit API is probably better to not touch the existing uses of `input`

Reply:
I did see this pattern in bedrock SDK. might be worth trying.

```
{
    message: string;
    required?: never;
    defaultValue?: string;
  } | {
    message: string;
    required: true;
    defaultValue?: never;
  }
```

Reply:
ohh that's neat. Updated.